genie-common-internal/src/main/java/com/netflix/genie/common/internal/configs/CommonServicesAutoConfiguration.java

```diff
@@ -83,14 +88,31 @@ public JobArchiveService jobArchiveService(
      * The manifest produced by this service do not include checksum for entries and caches manifests recently created.
      *
      * @param directoryManifestFactory the factory to produce the manifest if needed
+     * @param cache                    the cache to use
      * @return a {@link JobDirectoryManifestService}
      */
     @Bean
     @ConditionalOnMissingBean(JobDirectoryManifestService.class)
     public JobDirectoryManifestService jobDirectoryManifestService(
-        final DirectoryManifest.Factory directoryManifestFactory
+        final DirectoryManifest.Factory directoryManifestFactory,
+        @Qualifier("jobDirectoryManifestCache") final Cache<Path, DirectoryManifest> cache
     ) {
-        return jobDirectoryPath -> directoryManifestFactory.getDirectoryManifest(jobDirectoryPath, false);
+        return new JobDirectoryManifestServiceImpl(directoryManifestFactory, cache, false);
+    }
+
+    /**
+     * Provide a {@code Cache<Path, DirectoryManifest>} named "jobDirectoryManifestCache" if no override is defined.
+     *
+     * @return a {@link Cache}
+     */
+    @Bean(name = "jobDirectoryManifestCache")
+    @ConditionalOnMissingBean(name = "jobDirectoryManifestCache")
```

Comment:
Could also enforce the type but probably not super required

Reply:
I can't specify a bean type that includes generics. So by type it would just be `Cache`, which i feel would be too general. That's why i went with name.